lib/runner.js

```diff
@@ -1027,11 +989,80 @@ function extraGlobals() {
     });
 
     // 'errno' was renamed to process._errno in v0.9.11.
-
     if (nodeVersion < 0x00090b) {
       return ['errno'];
     }
   }
 
   return [];
 }
+
+/**
+ * {@link Runner}-related constants.
+ * @public
+ * @memberof Runner
+ * @readonly
+ * @enum {string}
+ */
+var constants = {
+  /**
+   * Emitted when {@link Hook} execution begins
+   */
+  EVENT_HOOK_BEGIN: 'hook',
+  /**
+   * Emitted when {@link Hook} execution ends
+   */
+  EVENT_HOOK_END: 'hook end',
+  /**
+   * Emitted when Root {@link Suite} execution begins (all files have been parsed and hooks/tests are ready for execution)
+   */
+  EVENT_RUN_BEGIN: 'start',
+  /**
+   * Emitted when Root {@link Suite} execution has been delayed via `delay` option
+   */
+  EVENT_RUN_DELAY: 'waiting',
```

Comment:
Think this makes the pairing any more obvious?
`EVENT_RUN_DELAY` => `EVENT_RUN_DELAY_BEGIN`
`EVENT_ROOT_SUITE_RUN` => `EVENT_RUN_DELAY_END`

Reply:
hmm, ok

Reply:
actually I can't move it, because the runner needs it, but I can add a new event to the runner which repeats it, so reporters can use it. ok.